apps/hash-frontend/src/pages/account/integrations/linear/new.page.tsx

```diff
@@ -1,100 +1,153 @@
 import { EntityId } from "@blockprotocol/graph/temporal";
+import { Button } from "@hashintel/design-system";
 import { types } from "@local/hash-isomorphic-utils/ontology-types";
-import { Entity } from "@local/hash-subgraph/.";
-import { getRoots } from "@local/hash-subgraph/stdlib";
 import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
-import { Container, Typography } from "@mui/material";
+import { Box, Container, Typography } from "@mui/material";
 import { useRouter } from "next/router";
-import { useEffect, useMemo, useState } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
 
-import { useBlockProtocolQueryEntities } from "../../../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-query-entities";
+import {
+  GetLinearOrganizationQuery,
+  GetLinearOrganizationQueryVariables,
+  SyncLinearIntegrationWithWorkspacesMutation,
+  SyncLinearIntegrationWithWorkspacesMutationVariables,
+} from "../../../../graphql/api-types.gen";
+import {
+  getLinearOrganizationQuery,
+  syncLinearIntegrationWithWorkspacesMutation,
+} from "../../../../graphql/queries/integrations/linear.queries";
 import {
   getLayoutWithSidebar,
   NextPageWithLayout,
 } from "../../../../shared/layout";
 import { useAuthenticatedUser } from "../../../shared/auth-info-context";
-import { SelectLinearTeams } from "./select-linear-teams";
+import {
+  LinearOrganizationTeamsWithWorkspaces,
+  mapLinearOrganizationToLinearOrganizationTeamsWithWorkspaces,
+  mapLinearOrganizationToSyncWithWorkspacesInputVariable,
+  SelectLinearTeamsTable,
+} from "../select-linear-teams-table";
+import { useLinearIntegrations } from "../use-linear-integrations";
 
 const NewLinearIntegrationPage: NextPageWithLayout = () => {
   const router = useRouter();
   const { authenticatedUser } = useAuthenticatedUser();
-  const { queryEntities } = useBlockProtocolQueryEntities();
 
-  const [syncedTeamsWithWorkspaces, setSyncedTeamsWithWorkspaces] =
-    useState<boolean>(false);
+  const [linearOrganization, setLinearOrganization] =
+    useState<LinearOrganizationTeamsWithWorkspaces>();
+
+  const [
+    syncLinearIntegrationWithWorkspaces,
+    { loading: loadingSyncLinearIntegrationWithWorkspaces },
+  ] = useMutation<
+    SyncLinearIntegrationWithWorkspacesMutation,
+    SyncLinearIntegrationWithWorkspacesMutationVariables
+  >(syncLinearIntegrationWithWorkspacesMutation);
+
+  const [getLinearOrganization] = useLazyQuery<
+    GetLinearOrganizationQuery,
+    GetLinearOrganizationQueryVariables
+  >(getLinearOrganizationQuery);
 
   const linearIntegrationEntityId = useMemo(() => {
     return router.query.linearIntegrationEntityId as EntityId;
   }, [router]);
 
-  const [linearIntegrationEntities, setLinearIntegrationEntities] =
-    useState<Entity[]>();
+  const { linearIntegrations } = useLinearIntegrations();
+
+  const linearIntegration = useMemo(() => {
```

Comment:
I think you could move this inside the fetching effect since you don't need it apart from to get the organization (the save callback can just use the `linearIntegrationEntityId` directly)

Reply:
See [this response](https://github.com/hashintel/hash/pull/2826#discussion_r1279043708) on why I'd prefer using `linearIntegration` than `linearIntegrationEntityId` directly. If you disagree happy to change this.

Reply:
Resolved in https://github.com/hashintel/hash/commit/0afd9f20d7568e28b60e92a2fa125ea628fec88a